### PR TITLE
Improved TPOT Handler to work with string data type ☕☕

### PR DESCRIPTION
**Pull request by parthiv11**
## Description
* Using LabelEncoder added support for str datatype in TPOT ML Handler

**Fixes** #4952

## Type of change

- [x] 🐛 Bug fix (non-breaking change which fixes an issue)

### CODE REVIEW

File: mindsdb/integrations/handlers/tpot_handler/tpot_handler.py [hunk 1]
```diff
@@ -33,8 +35,22 @@ def create(self, target: str, df: Optional[pd.DataFrame] = None, args: Optional[
         
 
         if df is not None:
+            # Separate out the categorical and non-categorical columns
+            categorical_cols = df.select_dtypes(include=['object']).columns.tolist()
```
[the remark this answers is not in the document]
Comment (parthiv11) [reply mindsdb/integrations/handlers/tpot_handler/tpot_handler.py]: Done

File: mindsdb/integrations/handlers/tpot_handler/tpot_handler.py [hunk 2]
```diff
@@ -33,8 +35,22 @@ def create(self, target: str, df: Optional[pd.DataFrame] = None, args: Optional[
         
 
         if df is not None:
+            # Separate out the categorical and non-categorical columns
+            categorical_cols=[col for col,type_col in type_of_cols.items() if type_col=='categorical']
```
Comment (paxcema) [inline mindsdb/integrations/handlers/tpot_handler/tpot_handler.py:39]: `from type_infer.dtype import dtype`

And then:

`if type_col in (dtype.categorical, dtype.binary)`

**Review by paxcema — CHANGES_REQUESTED** [review]

**Review by parthiv11 — COMMENTED** [review]
DOne

File: mindsdb/integrations/handlers/tpot_handler/tpot_handler.py (hunk 2 above)
Comment (parthiv11) [reply mindsdb/integrations/handlers/tpot_handler/tpot_handler.py]: this is not working

**Comment by parthiv11** [comment]
Done all things worked perfectly
Please checkout

**Review by paxcema — APPROVED** [review]

**Comment by parthiv11** [comment]
@ZoranPandovski since this PR is approved by @paxcema please merge it

**Review by ZoranPandovski — APPROVED** [review]

**Pull request merged**